openshift-knative-operator/pkg/serving/extension_test.go

```diff
@@ -35,14 +43,16 @@ func TestReconcile(t *testing.T) {
 	}
 
 	cases := []struct {
-		name     string
-		in       *v1alpha1.KnativeServing
-		objs     []runtime.Object
-		expected *v1alpha1.KnativeServing
+		name                    string
+		in                      *v1alpha1.KnativeServing
+		objs                    []runtime.Object
+		expected                *v1alpha1.KnativeServing
+		shouldDisableMonitoring bool
```

Comment:
I'd rather not introduce this specific concept to this test. Can we instead test more generic in that we make `expecetd` a list of `runtime.Object` and make sure that all the objects we expect are correctly reflected?

Reply:
I would like to signal that the env var for disabling monitoring should be set, this is external to the test case at hand and happens per test case not before tests are run, what is wrong with that? For the ns check at the end of the test I can add it to the expected objects. Do you mean the latter?

Reply:
I see. I'd check the env var behavior with a separate test as this conflates things quite a bit :thinking:

Reply:
ok will do.